src/Compute/Compute/Generated/Models/ComputeAutoMapperProfile.cs

```diff
@@ -100,12 +100,16 @@ private static void Initialize()
                 cfg.CreateMap<TO.PSVirtualMachineScaleSetVM, TO.PSVirtualMachineScaleSetVMList>();
                 cfg.CreateMap<FROM.Disk, FROM.DiskUpdate>();
                 cfg.CreateMap<FROM.DiskUpdate, FROM.Disk>();
-                cfg.CreateMap<FROM.DiskUpdate, TO.PSDiskUpdate>();
-                cfg.CreateMap<TO.PSDiskUpdate, FROM.DiskUpdate>();
+                cfg.CreateMap<FROM.DiskUpdate, TO.PSDiskUpdate>()
+                    .ForMember(c => c.Tags, o => o.Condition(r => (r.Tags != null)));
+                cfg.CreateMap<TO.PSDiskUpdate, FROM.DiskUpdate>()
+                    .ForMember(c => c.Tags, o => o.Condition(r => (r.Tags != null)));
                 cfg.CreateMap<FROM.Snapshot, FROM.SnapshotUpdate>();
                 cfg.CreateMap<FROM.SnapshotUpdate, FROM.Snapshot>();
-                cfg.CreateMap<FROM.SnapshotUpdate, TO.PSSnapshotUpdate>();
-                cfg.CreateMap<TO.PSSnapshotUpdate, FROM.SnapshotUpdate>();
+                cfg.CreateMap<FROM.SnapshotUpdate, TO.PSSnapshotUpdate>()
+                    .ForMember(c => c.Tags, o => o.Condition(r => (r.Tags != null)));
+                cfg.CreateMap<TO.PSSnapshotUpdate, FROM.SnapshotUpdate>()
+                    .ForMember(c => c.Tags, o => o.Condition(r => (r.Tags != null)));
```

Comment:
I'm a bit confused as to why this will fix the issue. The issue is that customers have Tag values that are being lost, so their Tags are filled and are being mapped to an empty tag. 
Now this condition checks if the customer Tag has values in it and maps it only if there are tag values? Isn't that the same behavior as before?

Reply:
@Sandido the issue is that in a PATCH call you only send items that need to be updated. If the tags are null, changing them to empty will be interpreted as Clearing out all tags. These conditions check to not map tags if they are null.

Reply:
@haagha if the end users do want to clear the tags, would they fail to achieve that?

Reply:
@vidai-msft The users can clear tags like normally before by passing an empty dictionary. An empty dictionary would not meet the condition.

Reply:
@haagha OK. got it. The expected behavior is empty dictionary will clear the tag while null should do nothing.